app/page/send.tsx

```diff
@@ -43,17 +75,51 @@ export function Send() {
 
   const handleSend = useCallback(async () => {
     const signer = await provider.getSigner()
-    const txResult = await signer.sendTransaction({
-      to: ethers.getAddress(txTo),
-      value: ethers.parseUnits(txValue.toString(), "ether")
+    const erc20 = getErc20Contract(token.address, provider)
+    const to = isEth ? txTo : token.address
+    const value = isEth ? ethers.parseUnits(txValue.toString(), "ether") : 0
+    const data = isEth
+      ? "0x"
+      : erc20.interface.encodeFunctionData("transfer", [
+          txTo,
+          ethers.parseUnits(txValue.toString(), token.decimals)
+        ])
+
+    await signer.sendTransaction({
+      to: to,
+      value: value,
+      data: data
     })
-    // TODO: Need to avoid Popup closure
-    setTxHash(txResult.hash)
   }, [txTo, txValue])
 
   return (
     <NavbarLayout>
       <div className="text-base justify-center items-center h-auto">
+        <div className="flex">
+          <label className="col-span-1">Asset:</label>
+          <select
+            id="asset"
+            value={token.address}
+            className="col-span-4 border w-full outline-none border-gray-300"
+            onChange={handleAssetChange}>
+            <option key={nativeToken.address} value={nativeToken.address}>
+              {nativeToken.symbol}: {nativeToken.balance} {nativeToken.symbol}
+            </option>
+            {tokens.map((token) => {
+              const balance = parseFloat(
+                ethers.formatUnits(
+                  ethers.toBeHex(token.balance),
+                  ethers.toNumber(token.decimals)
+                )
+              ).toFixed(6)
```

Comment:
nit: Why not using `formatUnitsToFixed` util?

Reply:
This was my oversight, let me fix it!